File: Basic_Concepts_of_Java/OOP.cpp
// Class  = Charcteristics + Behaviours
// Class  = Data + Functions

#include<iostream>
using namespace std;

class Demo
{  
    public:
        int No1;
        int No2;

        void fun()
        {
            cout<<"Inside fun\n";
        }
};

int main()
{
    Demo obj1;
    Demo obj2;

    cout<<"Size of object : "<<sizeof(obj1)<<"\n";  // 8

    obj1.No1 = 10;
    obj1.No2 = 20;

    obj2.No1 = 30;
    obj2.No2 = 40;
    
    obj1.fun(); // Inside fun
    obj2.fun(); // Inside fun

    cout<<obj1.No2<<"\n";   // 20

    return 0;
}

File: Basic_Concepts_of_Java/OOP.java
class OOP
{
    public static void main(String Arg[])
    {
        System.out.println("Inside main function");

        Arithematic obj1 = new Arithematic();
        Arithematic obj2 = new Arithematic(11,10);

        int iRet = 0;

        iRet = obj1.Addition();     //   0
        System.out.println("Addition is : "+iRet);

        iRet = obj1.Substraction();     //   0
        System.out.println("Substraction is : "+iRet);

        iRet = obj2.Addition();     //   21
        System.out.println("Addition is : "+iRet);

        iRet = obj2.Substraction();     //   1
        System.out.println("Substraction is : "+iRet);
    }
}

class Arithematic
{
    public int No1;
    public int No2;

    public Arithematic()    // Default constructor
    {
        System.out.println("Inside default constructor");
        this.No1 = 0;
        this.No2 = 0;
    }
    public Arithematic(int i, int j)    // Parametrised constructor
    {
        System.out.println("Inside parametrised constructor");
        this.No1 = i;
        this.No2 = j;
    }

    public int Addition()
    {
        int Ans= 0;
        Ans = this.No1 + this.No2;
        return Ans;
    }
    public int Substraction()
    {
        int Ans= 0;
        Ans = this.No1 - this.No2;
        return Ans;
    }
}